src/workloads/execution/TimeSeriesSortCompound.yml

```diff
@@ -0,0 +1,117 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This test exercises the behavior of the time series bounded sorter with a compound sort.
+
+GlobalDefaults:
+  dbname: &db test
+  coll: &coll Collection0
+  bucketsColl: &bucketsColl system.buckets.Collection0
+  batchSize: &batchSize 30000
+  fieldName: &field "numeric"
+  index: &index
+    keys: {numeric: 1}
+  nop: &Nop {Nop: true}
+
+Actors:
+- Name: CreateTimeSeriesCollection
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: *db
+    Operation:
+      OperationMetricsName: CreateTimeSeriesCollection
+      OperationName: RunCommand
+      OperationCommand:
+        {create: *coll, timeseries: {timeField: "t", metaField: "m"}}
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 1
+    Database: test
+    Operations:
+    - OperationName: RunCommand
+      OperationCommand:
+        drop: *coll
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - *Nop
+  - Repeat: 1
+    Database: *db
+    Collection: *coll
+    Threads: 1
+    CollectionCount: 1
+    DocumentCount: 10010000
+    BatchSize: *batchSize
+    Document:
+      t: {^Repeat: {count: 1000, fromGenerator: {^IncDate: {start: "2022-01-01", step: 36000}}}}
+      m: {^Cycle: {ofLength: 1000, fromGenerator: {^Inc: {start: 0}}}}
+  - *Nop
+  - *Nop
+  - *Nop
+
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *db
+  Phases:
+  - *Nop
+  - *Nop
+  - Repeat: 1
+  - *Nop
+  - *Nop
+
+- Name: Queries
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  - *Nop
+  - *Nop
+  - *Nop
+  - Repeat: 50
+    Database: *db
+    Operations:
+    - OperationMetricsName: SortCompound
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: *bucketsColl
+        pipeline: [
+          {$sort: {meta: 1, 'control.min.t': 1, 'control.max.t': 1}},
+          {$_internalUnpackBucket: {
+            exclude: [],
+            timeField: "t",
+            metaField: "m",
+            bucketMaxSpanSeconds: 3600,
+            assumeNoMixedSchemaData: true
+          }},
+          {$_internalBoundedSort: {
+            sortKey: {m: 1, t: 1},
+            bound: {base: "min"},
+          }},
+          {$_internalInhibitOptimization: {}},
+          {$skip: 1e10},
+        ]
+        cursor: {}
+        # allowDiskUse: true # TODO enable if committing without $_internal stages?
```

Comment:
Yea this seems right. Maybe enable it when we commit, and then when we're able to rewrite the blocking sort version of this into a bounded sort we can come back and disable it.